runtime/src/accounts_db.rs

```diff
@@ -753,6 +753,16 @@ impl RecycleStores {
     }
 }
 
+/// Removing unrooted slots in Accounts Background Service needs to be synchronized with flushing
+/// slots from the Accounts Cache.  This keeps track of those slots and the Mutex + Condvar for
+/// synchronization.
+#[derive(Debug, Default)]
+struct RemoveUnrootedSlotsSynchronization {
+    // slots being flushed from the cache or being purged
+    slots_under_contention: Mutex<HashSet<Slot>>,
+    signal: Condvar,
```

Comment:
fyi, our first usage of `Condvar` in the codebase. :)

Reply:
I know. I keep running into situations where this has almost been the piece I needed. Glad to know it is there.